src/analysis/bounded_variation.lean

```diff
@@ -634,6 +635,147 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
+  variation_on_from_to f s a b = (evariation_on f (s ∩ Icc a b)).to_real := if_pos h
+
+lemma variation_on_from_to_eq_of_ge {a b : α} (h : b ≤ a) :
+  variation_on_from_to f s a b = - (evariation_on f (s ∩ Icc b a)).to_real :=
+begin
+  dsimp only [variation_on_from_to],
+  split_ifs with h',
+  { cases le_antisymm h h',
+    rw [Icc_self, evariation_on.subsingleton, ennreal.zero_to_real],
+    simp only [neg_zero],
+    exact λ x hx y hy, hx.2.trans hy.2.symm, },
+  { refl, }
+end
+
+lemma variation_on_from_to_eq_neg_swap (a b : α) :
+  variation_on_from_to f s a b = - variation_on_from_to f s b a :=
+begin
+  rcases lt_trichotomy a b with ab|rfl|ba,
+  { simp only [variation_on_from_to, if_pos ab.le, if_neg ab.not_le, neg_neg], },
+  { simp only [variation_on_from_to_self, neg_zero], },
+  { simp only [variation_on_from_to, if_pos ba.le, if_neg ba.not_le, neg_neg], },
+end
+
+lemma variation_on_from_to_add (hf : has_locally_bounded_variation_on f s)
+  {a b c : α} (ha : a ∈ s) (hb : b ∈ s) (hc : c ∈ s) :
+  variation_on_from_to f s a b + variation_on_from_to f s b c = variation_on_from_to f s a c :=
+begin
+  rcases le_total a b with ab|ba;
+  rcases le_total a c with ac|ca;
+  rcases le_total b c with bc|cb,
+  any_goals
+  { cases le_antisymm (ab.trans bc) ca, cases le_antisymm ab bc,
+    simp only [variation_on_from_to_self, add_zero], },
+  any_goals
+  { cases le_antisymm (ba.trans ac) cb, cases le_antisymm ba ac,
+    simp only [variation_on_from_to_self, add_zero], },
+  any_goals { simp only [variation_on_from_to_eq_of_le ab] },
+  any_goals { simp only [variation_on_from_to_eq_of_le ac] },
+  any_goals { simp only [variation_on_from_to_eq_of_le bc] },
+  any_goals { simp only [variation_on_from_to_eq_of_ge ba] },
+  any_goals { simp only [variation_on_from_to_eq_of_ge ca] },
+  any_goals { simp only [variation_on_from_to_eq_of_ge cb] },
+  { rw ←ennreal.to_real_add (hf a b ha hb) (hf b c hb hc),
+    rw evariation_on.Icc_add_Icc f ab bc hb, },
+  { rw add_neg_eq_iff_eq_add,
+    rw ←ennreal.to_real_add (hf a c ha hc) (hf c b hc hb),
+    rw evariation_on.Icc_add_Icc f ac cb hc, },
+  { rw add_neg_eq_iff_eq_add, symmetry,
+    rw neg_add_eq_iff_eq_add,
+    rw [←ennreal.to_real_add (hf c a hc ha) (hf a b ha hb),
+        evariation_on.Icc_add_Icc f ca ab ha], },
+  { rw neg_add_eq_iff_eq_add,
+    rw ←ennreal.to_real_add (hf b a hb ha) (hf a c ha hc),
+    rw evariation_on.Icc_add_Icc f ba ac ha, },
+  { rw neg_add_eq_iff_eq_add, symmetry,
+    rw add_neg_eq_iff_eq_add,
+    rw ←ennreal.to_real_add (hf b c hb hc) (hf c a hc ha),
+    rw evariation_on.Icc_add_Icc f bc ca hc, },
+  { rw neg_add_eq_iff_eq_add, symmetry,
+    rw add_neg_eq_iff_eq_add, symmetry,
+    rw neg_add_eq_iff_eq_add,
+    rw ←ennreal.to_real_add (hf c b hc hb) (hf b a hb ha),
+    rw evariation_on.Icc_add_Icc f cb ba hb, },
+end
+
+lemma variation_from_monotone_on (hf : has_locally_bounded_variation_on f s) {a : α} (as : a ∈ s) :
```

Comment:
The order of words in the lemma name should follow the statement. So something like `monotone_on_variation_on_from_to` here. Same thing for the next lemmas.